lib/presentation/bottom_nav_screens/profile_screen/profile_screen.dart

```diff
@@ -1,17 +1,148 @@
-    return const Center(
-      child: Text('Profile'),
+    return Scaffold(
+      body: SingleChildScrollView(
+        child: Column(
+          children: [
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                IconButton(
+                  onPressed: () {},
+                  icon: const Icon(
+                    Icons.arrow_back,
+                  ),
+                ),
+                GestureDetector(
+                  onTap: () {},
+                  child: const Padding(
+                    padding: EdgeInsets.all(10),
+                    child: Text('Edit'),
+                  ),
+                )
+              ],
+            ),
+            const SizedBox(height: 40),
+            const CircleAvatar(
+              backgroundColor: Color(0xff173143),
+              radius: 70,
+            ),
+            const SizedBox(height: 20),
+            Text(
+              widget.name,
+              style: Theme.of(context).textTheme.headline6?.copyWith(
+                    fontWeight: FontWeight.w900,
+                  ),
+            ),
+            const SizedBox(height: 5),
+            Text(
+              widget.place,
+              style: Theme.of(context).textTheme.subtitle1?.copyWith(
+                    fontWeight: FontWeight.w500,
+                    color: AppColors.borderGrey,
+                  ),
+            ),
+            const SizedBox(height: 40),
+            Container(
+              height: 2,
+              color: AppColors.lightGrey,
+            ),
+            const SizedBox(height: 20),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceAround,
+              children: [
+                GestureDetector(
+                  onTap: () => context.push(RoutePaths.networkscreenRoute.path),
+                  child: Column(
+                    children: [
+                      const Icon(Icons.person_outline),
+                      const Text('Network'),
+                      Text(widget.network)
+                    ],
+                  ),
+                ),
+                GestureDetector(
+                  onTap: () => context.go('/reviewscreen'),
+                  child: Column(
+                    children: [
+                      const Icon(Icons.star_border_outlined),
+                      const Text('My Review'),
+                      Text(widget.reviews)
+                    ],
+                  ),
+                ),
+                GestureDetector(
+                  onTap: () => context.go('/reviewscreen'),
+                  child: Column(
+                    children: [
+                      Icon(
+                        Icons.emoji_events_outlined,
+                        color: foodietypeColor(widget.foodietype),
+                      ),
+                      const Text('My Level'),
+                      Text(
+                        (widget.foodietype == '1')
```

Comment:
I've seen this logic in multiple file, try to extract the logic in separate file so that we can access it globally( put inside core/utils)